ocrd_cis/ocropy/resegment.py

```diff
@@ -438,14 +424,15 @@ def _process_segment(self, parent, parent_image, parent_coords, page_id, zoom, l
                                                             parent_image, parent_coords)
                     other_polygon = polygon_for_parent(other_polygon, otherline.parent_object_)
                     if other_polygon is None:
-                        LOG.warning("Ignoring extant new polygon for line '%s'", otherline.id)
+                        self.logger.warning("Ignoring extant new polygon for line '%s'", otherline.id)
                         continue
                     otherline.get_Coords().set_points(points_from_polygon(other_polygon))
 
 def spread_dist(lines, old_labels, new_labels, binarized, components, coords,
-                maxdist=43, loc='', threshold=0.9):
+                maxdist=43, loc='', threshold=0.9, logger = None):
     """redefine line coordinates by contourizing spread of connected components propagated from new labels"""
-    LOG = getLogger('processor.OcropyResegment')
+    if not logger:
+        raise ValueError(f"Logger has not been passed by the caller")
```

Comment:
Then better make it a new positional argument instead of a kwarg.

Reply:
Resolved: [46d84d5](https://github.com/bertsky/ocrd_cis/pull/5/commits/46d84d58b7474adc3cb9f9b756b215efebd495e3)